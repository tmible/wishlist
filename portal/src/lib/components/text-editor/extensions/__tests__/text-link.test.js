import { describe, expect, it, vi } from 'vitest';
import TextLink from '../text-link.js';

describe('text editor / extensions / text link', () => {
  it('should be defined', () => {
    expect(TextLink).toMatchSnapshot();
  });

  describe('add attributes', () => {
    it('should add attributes', () => {
      expect(TextLink.config.addAttributes()).toMatchSnapshot();
    });

    it('should parse href from HTML', () => {
      const getAttribute = vi.fn();
      TextLink.config.addAttributes().href.parseHTML({ getAttribute });
      expect(getAttribute).toHaveBeenCalledWith('href');
    });
  });

  it('should parse HTML', () => {
    expect(TextLink.config.parseHTML()).toMatchSnapshot();
  });

  it('should render HTML', () => {
    const bindValue = { options: { HTMLAttributes: { 'from-options': 'attribute' } } };
    const argument = { HTMLAttributes: { 'from-argument': 'attribute' } };
    expect(TextLink.config.renderHTML.bind(bindValue)(argument)).toMatchSnapshot();
  });

  describe('add commands', () => {
    it('should add set command', () => {
      const setMark = vi.fn();
      TextLink.config.addCommands().setTextLink('attributes')({ commands: { setMark } });
      expect(setMark).toHaveBeenCalledWith(TextLink.name, 'attributes');
    });

    it('should add toggle command', () => {
      const toggleMark = vi.fn();
      TextLink.config.addCommands().toggleTextLink('attributes')({ commands: { toggleMark } });
      expect(toggleMark).toHaveBeenCalledWith(TextLink.name, 'attributes');
    });

    it('should add unset command', () => {
      const unsetMark = vi.fn();
      TextLink.config.addCommands().unsetTextLink()({ commands: { unsetMark } });
      expect(unsetMark).toHaveBeenCalledWith(TextLink.name);
    });
  });
});
