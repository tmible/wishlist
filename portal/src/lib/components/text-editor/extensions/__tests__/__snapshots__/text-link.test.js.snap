// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`text editor / extensions / text link > add attributes > should add attributes 1`] = `
{
  "class": {
    "default": "cursor-pointer",
  },
  "href": {
    "default": "",
    "parseHTML": [Function],
  },
}
`;

exports[`text editor / extensions / text link > should be defined 1`] = `
Mark {
  "child": null,
  "config": {
    "addAttributes": [Function],
    "addCommands": [Function],
    "defaultOptions": {},
    "name": "text_link",
    "parseHTML": [Function],
    "renderHTML": [Function],
  },
  "name": "text_link",
  "options": {},
  "parent": null,
  "storage": {},
  "type": "mark",
}
`;

exports[`text editor / extensions / text link > should parse HTML 1`] = `
[
  {
    "tag": "a[data-entity="text_link"]",
  },
]
`;

exports[`text editor / extensions / text link > should render HTML 1`] = `
[
  "a",
  {
    "from-argument": "attribute",
    "from-options": "attribute",
  },
  0,
]
`;
